src/caffe/layers/data_layer.cpp

```diff
@@ -112,11 +118,12 @@ void DataLayer<Dtype>::load_batch(Batch<Dtype>* batch) {
   for (int item_id = 0; item_id < batch_size; ++item_id) {
       timer.Start();
       // Get a datum
-      Datum* datum = (reader_.full().pop("Waiting for data"));
+      string* str = (reader_.full().pop("Waiting for data"));
```

Comment:
You fail out (as the parallel code always has): https://github.com/NVIDIA/caffe/blob/caffe-0.14/src/caffe/parallel.cpp#L454